File: javascript/bob/bob.js
//
// This is only a SKELETON file for the "Bob" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

var Bob = function() {};

Bob.prototype.hey = function(input) {
  var hasLetters = input.match(/[a-z]+/i)
  var isShouting = input == input.toUpperCase();
  var isQuestioning = input.slice(-1) == '?';
  var isEmpty = input.trim().length == 0

  if(hasLetters && isShouting) {
    return "Whoa, chill out!";
  } else if(isQuestioning) {
    return "Sure.";
  } else if(isEmpty) {
    return "Fine. Be that way!";
  } else {
    return "Whatever.";
  }
};

module.exports = Bob;

File: ruby/bob/bob.rb
class Bob
  def hey(remark)
    return "Fine. Be that way!" if remark.strip.empty?
    return "Whoa, chill out!" if remark.upcase == remark && remark.downcase != remark
    return "Sure." if remark.end_with?("?")
    "Whatever."
  end
end
